apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessagePropertyAccessor.java

```diff
@@ -61,9 +64,9 @@ public String getFirstHeader(String headerName, Message message) {
 
     @Nonnull
     private String jmsifyHeaderName(String headerName) {
-        if (headerName.equals(TraceContext.ELASTIC_TRACE_PARENT_TEXTUAL_HEADER_NAME)) {
+        if (tracer.getTraceParentHeaders(TraceHeaderDisplay.REGULAR).contains(headerName)) {
```

Comment:
Wouldn't it make more sense to entirely remove the `jmsifyHeaderName` method and logic here and instead to add a `TraceHeaderDisplay` parameter to the `AbstractSpan.propagateContext` method ?

Reply:
Okay, alternative thought:

What would you think about pushing the responsibility of encoding the header names in general (`TraceHeaderDisplay `) to the `HeaderSetter` and `HeaderGetter`s?
This would mean that you can (and should) remove the `TraceHeaderDisplay` parameter from all `Tracer` methods except for `getTraceParentHeaders`. I think this approach would avoid bugs such as accidentally using `TraceHeaderDisplay.REGULAR` for kafka, like in your PR.

In addition it would make it easier in the future to remove the `TraceHeaderDisplay` enum when we get rid of `elastic-apm-tracer`-header support.